### PR TITLE
Clean up context / 1.0.0 development

### PR DESCRIPTION
**Pull request by KCErb**
Whew that was some work there that was.

There's a lot of changes here so it'll need careful review / testing. Based on my tests this preserves functionality, cleans up the context, makes it a lot more CDQ / ActiveRecord like and paves the way for doing migrations in the future!

A major breaking change here (i.e. full version bump if we keep it) is that I huddled everything into the `SwissDB` namespace. That means a model has to inherit from `SwissDB::SwissModel` instead of just `SwissModel`. 

I decided to include that in this pull request for two reasons:
1. Cleaning up the context like this means polluting their namespace with only one constant: `SwissDB`. We've got to tuck everything else inside of this.
2. There's already so much new here that a version bump is appropriate, may as well introduce a breaking change.

On this PR I think I should probably add some stuff to the README and maybe a wiki page on schemas since this PR makes a schema's version usable / important.

Here's what's new
- `SwissDB` is now our top-level namespace and as such holds onto things we want classes to share like the context and the data store.
- schema and version are now written into the `resources/raw` folder during the rake process and then read back into the app when we need them.
- A schema's version will need to follow the android convention of being convertible to an integer i.e. `001` or `42`. I'll be using this heavily in the next project: basic migration support.
- I "cleaned up" a lot of things as I went. That means I changed style sometimes when I thought it would help.

Please give this all a careful read and ask me all the questions you have!

### CODE REVIEW

**Comment by mixflame** [comment]
The breaking change is fine. We can alert our users to it in the README. They will find it there. It makes sense anyway in the global goal of reducing namespace pollution. Code looks great. If it works, awesome. I will test it out shortly (excited!). If there's a new call `SwissDB.setup` it should be documented in a README section explaining where to put it and what it does. Also in theory the library should work in all RubyMotion Android apps not just BluePotion. Good stuff here. :D :zap:

**Comment by mixflame** [comment]
I tried it but no success. I get

```
E/com/company/app(11675): Exception raised: NameError: uninitialized constant `SwissDB'
```

when I tried to do

```
class Model < SwissDB::SwissModel
```

I can't find anything saying that using modules as namespaces is unsupported but I think I've had problems with it.

This works for you?

**Comment by mixflame** [comment]
Update: According to @bmichotte from the #android room on Slack, it's not supported (yet) (modules as namespaces for classes). We will have to drop the SwissDB namespace. We could prefix "Swiss" to all class names though like SwissDataStore, SwissModel, SwissCursor, whatever...

**Comment by mixflame** [comment]
Update: This seems to be an error in my own code because even taking the namespace out doesn't work.

Update 2: No, this is related to this pull request. Something about it prevents it from being required on my end.

**Comment by mixflame** [comment]
I'm making progress here

**Comment by KCErb** [comment]
Hey @jsilverMDX, sorry to hear this isn't going smoothly.
- I'd be very sorry to drop the namespace until we see something break from it, but ultimately you're the project manager on this one so I'll drop it if you like.
- Sorry that I forgot to mention it looks for `schema.rb`. We don't have to keep that, but we do need a way to uniquely identify which schema file should be used in a call to `onCreate`.
- Everything _does_ work in my test app but this just speaks to the need to write up some tests :)

**Comment by mixflame** [comment]
It's not going that bad. There were a few things. We don't have to drop the namespace. The schema.rb convention is fine for now. Right now the only problem is that my datastore is nil. I gotta figure out why that is.

**Comment by mixflame** [comment]
I think I've managed to solve #24 in this branch. :+1: We should have no more leaky cursors problem. I will test it out now (stability wise. code wise they are now closed properly.)

File: lib/swiss_db/swiss_model.rb [hunk 1]
```diff
@@ -1,121 +1,140 @@
...
+      end
 
-  def self.destroy_all!
-    # destroy all of this kind (empty table)
-    store.destroy_all(@table_name)
-  end
+      def first
+        # select all and get first
+        cursor = all.first
+        cursor
```
Comment (KCErb) [inline lib/swiss_db/swiss_model.rb]: this should be called `model` not cursor. `model#all` does return a cursor but `all.first` returns a new instance of the model.

Comment (mixflame) [inline lib/swiss_db/swiss_model.rb]: its fixed now

**Comment by mixflame** [comment]
Good news, this is fixed and working as of motion 4.11.

**Comment by mixflame** [comment]
I'm going to merge and release this as it's stable and working now. We can shoot for closing the issues in the next build.

**Comment by KCErb** [comment]
Woohoo!!! Great to hear!

**Pull request merged**